Firestore/Example/Tests/Remote/FSTRemoteEventTests.mm

```diff
@@ -420,41 +503,98 @@ - (void)testExistenceFilterMismatchResetsTarget {
 
-  XCTAssertEqual(event.targetChanges.count, 1);
+  XCTAssertEqual(event.targetChanges.size(), 2);
+
+  FSTTargetChange *targetChange1 = FSTTestTargetChange(
+      DocumentKeySet{}, DocumentKeySet{doc1.key, doc2.key}, DocumentKeySet{}, _resumeToken1, YES);
+  XCTAssertEqualObjects(event.targetChanges.at(1), targetChange1);
+
+  FSTTargetChange *targetChange2 =
+      FSTTestTargetChange(DocumentKeySet{}, DocumentKeySet{}, DocumentKeySet{}, _resumeToken1, NO);
+  XCTAssertEqualObjects(event.targetChanges.at(2), targetChange2);
+
+  // The existence filter mismatch will remove the document from target 1,
+  // but not synthesize a document delete.
+  FSTExistenceFilterWatchChange *change4 =
+      [FSTExistenceFilterWatchChange changeWithFilter:[FSTExistenceFilter filterWithCount:1]
+                                             targetID:1];
+  [aggregator handleExistenceFilter:change4];
 
-  FSTUpdateMapping *mapping1 =
-      [FSTUpdateMapping mappingWithAddedDocuments:@[ doc1, doc2 ] removedDocuments:@[]];
-  XCTAssertEqualObjects(event.targetChanges[@1].mapping, mapping1);
-  XCTAssertEqual(event.targetChanges[@1].snapshotVersion, testutil::Version(3));
-  XCTAssertEqual(event.targetChanges[@1].currentStatusUpdate, FSTCurrentStatusUpdateMarkCurrent);
-  XCTAssertEqualObjects(event.targetChanges[@1].resumeToken, _resumeToken1);
+  event = [aggregator remoteEventAtSnapshotVersion:testutil::Version(4)];
 
-  [event handleExistenceFilterMismatchForTargetID:@1];
+  FSTTargetChange *targetChange3 = FSTTestTargetChange(
+      DocumentKeySet{}, DocumentKeySet{}, DocumentKeySet{doc1.key, doc2.key}, [NSData data], NO);
+  XCTAssertEqualObjects(event.targetChanges.at(1), targetChange3);
 
-  // Mapping is reset
-  XCTAssertEqualObjects(event.targetChanges[@1].mapping, [[FSTResetMapping alloc] init]);
-  // Reset the resume snapshot
-  XCTAssertEqual(event.targetChanges[@1].snapshotVersion, testutil::Version(0));
-  // Target needs to be set to not current
-  XCTAssertEqual(event.targetChanges[@1].currentStatusUpdate, FSTCurrentStatusUpdateMarkNotCurrent);
-  XCTAssertEqual(event.targetChanges[@1].resumeToken.length, 0);
+  XCTAssertEqual(event.targetChanges.size(), 1);
+  XCTAssertEqual(event.targetMismatches.size(), 1);
+  XCTAssertEqual(event.documentUpdates.size(), 0);
+}
+
+- (void)testExistenceFilterMismatchRemovesCurrentChanges {
+  NSDictionary<FSTBoxedTargetID *, FSTQueryData *> *targetMap = [self listensForTargets:@[ @1 ]];
+
+  FSTDocument *doc1 = FSTTestDoc("docs/1", 1, @{ @"value" : @1 }, NO);
+
+  FSTDocumentWatchChange *addDoc = [[FSTDocumentWatchChange alloc] initWithUpdatedTargetIDs:@[ @1 ]
+                                                                           removedTargetIDs:@[]
+                                                                                documentKey:doc1.key
+                                                                                   document:doc1];
+
+  FSTWatchTargetChange *markCurrent =
```

Comment:
The order here seems mixed up. You've got
  * new change 1
  * new change 2
  * new aggregator
  * use change 1
  * use change 2
  * new change 3
  * use change 3

I suggest creating all the changes first then pushing them through the aggregator to make it obvious what the sequence of changes is. Alternatively, create and use right away as you've done below.

Reply:
I have changed all test cases to declare the variables much closer to where they are used. The new "layout" is now "create doc, create change, apply change". That should be cleaner than what I had before.